app/infra/repositories/picture/base.py

```diff
@@ -1,20 +1,34 @@
 from abc import ABC, abstractmethod
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from io import BytesIO
+from uuid import uuid4
 
 
 @dataclass
-class Picture:
+class PictureObject:
     obj: BytesIO
-    public_link: str = ''
+    ext: str
+    uid: str = field(default_factory=lambda: str(uuid4()))
+
+    @property
+    def name(self) -> str:
+        return f'{self.uid}.{self.ext}'
+
+    @property
+    def size(self) -> int:
+        return self.obj.getbuffer().nbytes
 
 
 @dataclass
 class BasePictureRepository(ABC):
     @abstractmethod
-    def get(self, identifier) -> Picture:
+    def create(self, filename: PictureObject) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def read(self, filename: str) -> PictureObject:
         raise NotImplementedError()
 
     @abstractmethod
-    def get_random(self) -> Picture:
+    def delete(self, filename: str) -> None:
```

Comment:
Why do you include a delete method to a BasePictureRepository class?

Reply:
It might be a bad decision to change the purpose of a class on a fly - somebody else might already be using your solution for his/her tasks

Reply:
I have added it to imitate CRUD complied repository (but without `update` because of lack of time and for sick of simplicity, but still it is easy to implement `update` part so it become CRUD-full repository). 

About second comment, yes, it might be a problem, but actually it is simply renaming, `BasePictureRepository` -> `BasePictureSource`, because I have considered it to be a bad naming. Also it is not considered to be a library or package, but complete application, so very unlikely anybody could reuse code from it for their own tasks outside of this codebase itself, so there is only one problem IMO, it could lead to merge conflicts if different devs would work on this part in parallel. Such a renaming should be discussed in advance in team and done in advance, so merge conflict will be avoided or easily resolvable.

Reply:
Very unlikely -> still could happen, especially in case of interfaces. Just be careful with this during real projects work